File: benchmarks/julia/src/recursion_fibonacci.jl
using BenchmarkTools
using Test

include("utils.jl")

recursion_fibonacci(n) = n < 2 ? n : recursion_fibonacci(n - 1) + recursion_fibonacci(n - 2)

# Test output
@test recursion_fibonacci(20) == 6765

# Run benchmark
write_benchmark_result("recursion_fibonacci", @benchmark recursion_fibonacci(20))

File: benchmarks/javascript/src/recursion_fibonacci.js
import { strict as assert } from "node:assert";
import { benchmark, write_benchmark_result } from "./utils.js";

const recursion_fibonacci = (n) =>
  n < 2 ? n : recursion_fibonacci(n - 1) + recursion_fibonacci(n - 2);

// Test output
assert(recursion_fibonacci(20) == 6765);

// Run benchmark
let times = benchmark(recursion_fibonacci, [20], 1000, 100);
write_benchmark_result("recursion_fibonacci", times);
